lib/ansible/module_utils/basic.py

```diff
@@ -945,16 +946,19 @@ def set_default_selinux_context(self, path, changed):
         context = self.selinux_default_context(path)
         return self.set_context_if_different(path, context, False)
 
-    def set_context_if_different(self, path, context, changed, diff=None):
+    def set_context_if_different(self, path, context, changed, diff=None, expand=True):
 
         if not HAVE_SELINUX or not self.selinux_enabled():
             return changed
-        cur_context = self.selinux_context(path)
+        if expand:
+            path = os.path.expanduser(os.path.expandvars(path))
+        b_path = to_bytes(path, errors='surrogate_or_strict')
+        cur_context = self.selinux_context(b_path)
```

Comment:
Ok, I am also wondering why I am adding the expand-option. Probably to keep things consistent, but if nobody complained about not expanding, why should I care, right ? :-)

Will revert.